src/cppProperties.ts

```diff
@@ -0,0 +1,215 @@
+
+    if (fs.existsSync(path.join(workspaceFolder, cleanupFile))) {
+        await executeRCommand(`system("sh ./${cleanupFile}")`, workspaceFolder, (e: Error) => {
+            void window.showErrorMessage(e.message);
+            return '';
+        });
+    }
+
+    const compileInfo = extractCompilerInfo(compileOutputCpp);
+    const compileStdCpp = extractCompilerStd(compileOutputCpp);
+    const compileStdC = extractCompilerStd(compileOutputC);
+    const compileCall = extractCompilerCall(compileOutputCpp);
+    const compilerPath = await executeRCommand(`cat(Sys.which("${compileCall}"))`, workspaceFolder, (e: Error) => {
+        void window.showErrorMessage(e.message);
+        return '';
+    });
+
+    const intelliSensePlatform = platformChoose('windows', 'macos', 'linux');
+    const intelliSenseComp = compileCall.includes('clang') ? 'clang' : 'gcc';
+    const intelliSense = `${intelliSensePlatform}-${intelliSenseComp}-${process.arch}`;
+
+    // Collect information from 'DESCRIPTION'
+    const linkingToIncludes = await collectRLinkingTo(workspaceFolder);
+
+    // Combine information
+    const envIncludes: string[] = ['${workspaceFolder}/src'];
+    envIncludes.push(...compileInfo.compIncludes.map((v) => path.isAbsolute(v) ? v : `\${workspaceFolder}/${path.join('src', v)}`));
+    envIncludes.push(...linkingToIncludes);
+
+    const envDefines = compileInfo.compDefines;
+
+    // If no standard is set on linux, the C standard seems to default to the c++ one.
+    const envCStd = compileStdC.includes('++') ? '${default}' : compileStdC;
+
+    const platformName = platformChoose('Win32', 'Mac', 'Linux');
+
+    // Build json
+    const re = {
+        'configurations': [{
+            'name': platformName,
+            'defines': envDefines,
+            'includePath': envIncludes,
+            'compilerPath': compilerPath,
+            'cStandard': envCStd,
+            'cppStandard': compileStdCpp,
+            'intelliSenseMode': intelliSense
+        }],
+        'version': 4
+    };
+    const ser = JSON.stringify(re, null, 2);
+
+    // Write file
+    const vscodeDir = path.join(workspaceFolder, '.vscode');
+    if (!fs.existsSync(vscodeDir)) {
+        fs.mkdirSync(vscodeDir);
+    }
+    fs.writeFileSync(path.join(vscodeDir, 'c_cpp_properties.json'), ser);
+}
+
+async function collectRLinkingTo(workspaceFolder: string): Promise<string[]> {
+    if (!fs.existsSync(path.join(workspaceFolder, 'DESCRIPTION'))) {
+        return [];
+    }
+
+    const rScript = extensionContext.asAbsolutePath('R/cppproperties/extractLinkingTo.R').replace(/\\/g, '/');
+    const linkingToIncludesStr = await executeRCommand(`source('${rScript}')`, workspaceFolder, (e: Error) => {
+        void window.showErrorMessage(e.message);
+        return '';
+    });
+    if (linkingToIncludesStr === '') {
+        return [];
+    }
+    return linkingToIncludesStr.split('////');
```

Comment:
Just a small nitpick: `////` is rendered as following by "Better Comments":

<img width="450" alt="image" src="https://user-images.githubusercontent.com/4662568/193159048-a690ac5e-142f-4c77-8333-b0dbaa5a1435.png">

Do we have a good reason to use such instead of a simple one like `\n`?

Reply:
Yes, I used this as a delimiter as newline characters are technically allowed in filenames in POSIX (depending on file system). 
'/' is the only disallowed one. We cant use '//' as it is used in Windows disk and network drives (i.e.: 'C://Users/...'). '///' might work, but I was unsure if there was an edge case with windows network drives where this might happen, so I chose '////'.
(See: https://stackoverflow.com/questions/1976007/what-characters-are-forbidden-in-windows-and-linux-directory-names)

"Better Comments" seems to have many open issues about false-positive comment detection inside strings (e.g.: https://github.com/aaron-bond/better-comments/issues/308 , https://github.com/aaron-bond/better-comments/issues/417 , https://github.com/aaron-bond/better-comments/issues/346)

Reply:
Looks like `/usr////local////lib` might be also a legal path?

Reply:
The files are now URL encoded and seperated by newlines: 9d8ad26
This should be robust against bugs caused by any special characters.